File: Chap6-GraphicsViewFramework/6-9ResizableItemIterface/resizablepixmapitem.cpp
#include "resizablepixmapitem.h"
#include <QPen>
#include <QGraphicsScene>
#include <QPainter>

ResizablePixmapItem::ResizablePixmapItem(QPixmap pixmap) :
    mPixmap(pixmap)
{
     setRect (QRectF(10,10,300,300));
     setOwnerItem(this);
}

QRectF ResizablePixmapItem::selectorFrameBounds() const
{
    return  rect();
}

void ResizablePixmapItem::setSelectorFrameBounds(QRectF boundsRect)
{

    prepareGeometryChange();
    setRect(boundsRect);
    update();


}

QRectF ResizablePixmapItem::boundingRect() const
{
    return selectorFrameBounds();
}

void ResizablePixmapItem::paint(QPainter *painter, const QStyleOptionGraphicsItem *option, QWidget *widget)
{
    Q_UNUSED(option);
    Q_UNUSED(widget);
    painter->save();

//    painter->setBrush(brush());

//    painter->drawRect(rect());
    painter->drawPixmap(boundingRect(),mPixmap,mPixmap.rect());

    drawHandlesIfNecessary();

    painter->restore();
}





QPixmap ResizablePixmapItem::getPixmap() const
{
    return mPixmap;
}

void ResizablePixmapItem::setPixmap(const QPixmap &value)
{
    mPixmap = value;
}

File: Chap6-GraphicsViewFramework/6-10RenderSceneToImageDemo/resizablepixmapitem.py
from PySide6.QtWidgets import QGraphicsRectItem
from PySide6.QtCore import QRectF
from PySide6.QtGui import QPainter, QPixmap

from resizablehandlerect import ResizableHandleRect

class ResizablePixmapItem(QGraphicsRectItem, ResizableHandleRect):
    """
    A pixmap item that can be resized using handles on its corners.
    """
    def __init__(self, pixmap):
        """
        Initialize the resizable pixmap item with the given pixmap.
        
        Args:
            pixmap: QPixmap to display
        """
        QGraphicsRectItem.__init__(self)
        ResizableHandleRect.__init__(self)
        self.m_pixmap = pixmap
        self.setRect(QRectF(10, 10, 300, 300))
        self.setOwnerItem(self)
    
    def boundingRect(self):
        """
        Get the bounding rectangle of the item.
        
        Returns:
            QRectF: The bounding rectangle
        """
        return self.selectorFrameBounds()
    
    def paint(self, painter, option, widget):
        """
        Paint the item.
        
        Args:
            painter: QPainter
            option: QStyleOptionGraphicsItem
            widget: QWidget
        """
        painter.save()
        painter.drawPixmap(self.boundingRect(), self.m_pixmap, self.m_pixmap.rect())
        self.drawHandlesIfNecessary()
        painter.restore()
    
    def selectorFrameBounds(self):
        """
        Get the bounds of the selector frame.
        
        Returns:
            QRectF: The bounds rectangle
        """
        return self.rect()
    
    def setSelectorFrameBounds(self, bounds_rect):
        """
        Set the bounds of the selector frame.
        
        Args:
            bounds_rect: QRectF with the new bounds
        """
        self.prepareGeometryChange()
        self.setRect(bounds_rect)
        self.update()
    
    def getPixmap(self):
        """
        Get the pixmap.
        
        Returns:
            QPixmap: The pixmap
        """
        return self.m_pixmap
    
    def setPixmap(self, pixmap):
        """
        Set the pixmap.
        
        Args:
            pixmap: QPixmap to display
        """
        self.m_pixmap = pixmap
        self.update()